templates/repo/diff/box.tmpl

```diff
@@ -31,12 +31,57 @@
 			{{end}}
 			{{template "repo/diff/whitespace_dropdown" .}}
 			{{template "repo/diff/options_dropdown" .}}
+			{{if .PageIsPullFiles}}
+				<script id="commit-data-script">
+					{{$numCommits := len .Commits}}
+					window.config.pageData.commitInfo = {
+						issueLink: {{$.Issue.Link}},
+						lastReviewCommitSha: {{.LastReviewSha}},
+						commits: [
+						{{range .Commits}}
+								{
+									summary: {{.Summary}},
+									committerOrAuthorName: {{if .Committer}}{{.Committer.Name}}{{else}}{{.Author.Name}}{{end}},
+									id: {{.ID.String}},
+									time: {{if .Committer}}{{.Committer.When}}{{else}}{{.Author.When}}{{end}},
+									selected: false
+								},
+						{{end}}],
+						locale: {
+							lang: {{$.locale}},
+							filter_changes_by_commit: {{.locale.Tr "repo.pulls.filter_changes_by_commit"}},
+							show_all_commits: {{.locale.Tr "repo.pulls.show_all_commits"}},
+							stats_num_commits: {{.locale.TrN $numCommits "repo.activity.git_stats_commit_1" "repo.activity.git_stats_commit_n" $numCommits}},
+							show_changes_since_your_last_review: {{.locale.Tr "repo.pulls.show_changes_since_your_last_review"}},
+						},
+						queryParams: '?style={{if $.IsSplitStyle}}split{{else}}unified{{end}}&whitespace={{$.WhitespaceBehavior}}&show-outdated={{$.ShowOutdatedComments}}'
+					};
+				</script>
+				<div id="diff-commit-select">
+					{{/*
+						the following will be replaced by vue component
+						but this avoids any loading artifacts till the vue component is initialized
+					*/}}
+					<div class="ui jump dropdown basic button custom">
+						{{svg "octicon-git-commit"}}
+					</div>
+				</div>
+			{{end}}
 			{{if and .PageIsPullFiles $.SignedUserID (not .IsArchived)}}
 				{{template "repo/diff/new_review" .}}
 			{{end}}
 		</div>
 	</div>
 	{{if not .DiffNotAvailable}}
+		{{if and .IsShowingOnlySingleCommit .PageIsPullFiles}}
+			<div class="ui info message">
+				<div class="text gt-whitespace-pre">{{.locale.Tr "repo.pulls.showing_only_single_commit" (ShortSha .BeforeCommitID)}}</div>
```

Comment:
Why `<div class="text gt-whitespace-pre">`? The message doesn't seem special

Reply:
this is / was just a copy paste - i've removed it